pandas/core/arrays/categorical.py

```diff
@@ -1341,6 +1341,8 @@ def searchsorted(self, value, side='left', sorter=None):
 
         if -1 in values_as_codes:
             raise ValueError("Value(s) to be inserted must be in categories.")
+        if is_scalar(value):
```

Comment:
would rather do this in ``pandas/core/base.py/searchsorted``

use ``is_scalar`` rather than a numpy function

Reply:
* the issue is rather with the helper function `_get_codes_for_values` which always returns an array. I didn't want to change it there since the way it is written right now only works for array like objects. In `base.py` we're already calling `searchsorted` directly on the numpy array, i.e. it obeys the in/output shape
* I'm using `is_scalar` here, is this wrong? Are you referring to the `np.asscalar`? I couldn't find a suitable pandas function for that (other than ~ `values[0]`)

Reply:
As I found out in #21699, numpy.searchsorted doesn't like python ints, but needs numpy ints to archieve its speed.

```python
>>> n = 1_000_000
>>> c = pd.Categorical(list('a' * n + 'b' * n + 'c' * n), ordered=True)
>>> %timeit c.codes.searchsorted(1)  # python int
7 ms ± 24.7 µs per loop
>>> c.codes.dtype
int8
>>> %timeit c.codes.searchsorted(np.int8(1))
2.46 µs ± 82.4 ns per loop
```

So the scalar version should be ``values_as_codes = values_as_codes[0]`` to avoid speed loss.